continuous_integration.md

```diff
@@ -6,147 +6,174 @@ done with the corresponding commands `deno test`, `deno lint` and `deno fmt`. In
-`.lcov` format, which you could then upload to one of the various code coverage
-platforms available on the Web:
+      # Run all test files in the repository and collect code coverage. The example
+      # runs with all permissions, but it is recommended to run with the minimal permissions your program needs (for example --allow-read).
+      - run: deno test --allow-all --coverage cov
 
-```yaml
-- name: Generate coverage report
-  run: deno coverage --lcov cov > cov.lcov
+      # This generates a report from the collected coverage in `deno test --coverage`. It is
+      # stored as a .lcov file which integrates well with services such as Codecov, Coveralls and Travis CI.
+      - run: deno coverage --lcov cov > cov.lcov
 ```
 
 ### Cross-platform workflows
 
 As a Deno module maintainer, you probably want to know that your code works on
 all of the major operating systems in use today: Linux, MacOS and Windows. A
-Cross-platform workflow can be achieved by running a matrix of parallel jobs in
-GitHub Actions, each one running your build on a different operating system:
+cross-platform workflow can be achieved by running a matrix of parallel jobs,
+each one running the build on a different OS:
 
 ```yaml
 jobs:
   build:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ ubuntu-latest, macos-latest, windows-latest ]
+        os: [ ubuntu-20.04, macos-11, windows-2019 ]
     steps:
-      # build goes here
+      # Build goes here.
 ```
 
-> Note: GitHub Actions has a known issue with handling Windows-style line
-> endings (CRLF). This may cause issues when running `deno fmt` in a pipeline
-> with jobs that run on `windows`. To solve this, configure the Actions runner
-> to use Linux-style line-endings with `git config --system core.autocrlf false`
-> and `git config --system core.eol lf` before running
-> `uses: actions/checkout@v2` in the pipeline.
+> Note: GitHub Actions has a known
+> [issue](https://github.com/actions/checkout/issues/135) with handling
+> Windows-style line endings (CRLF). This may cause issues when running
+> `deno fmt` in a pipeline with jobs that run on `windows`. To prevent this,
+> configure the Actions runner to use Linux-style line endings before running
+> the `actions/checkout@v2` step:
+>
+> ```
+> git config --system core.autocrlf false
+> git config --system core.eol lf
+> ```
+
+If you are working with experimental or unstable Deno APIs, you can include a
+matrix job running the canary version of Deno. This can help to spot breaking
+changes early on:
 
-There can be parts of the pipeline that don't make sense to run for every OS.
-For example, generating the same coverage report on Linux, MacOS and Windows is
-a bit redundant. You can use the conditional `if` keyword in these cases to
-reduce repetition:
+```yaml
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }} # Continue in case the canary run does not succeed
```

Comment:
```suggestion
    continue-on-error: ${{ matrix.canary }} # Continue in case the canary run does not succeed
```

Reply:
Or is `experimental` required?

Reply:
`canary` works too, I think `experimental` is just what the GitHub Actions docs uses